File: LeetCode_ProblemSet/Go/Problem2053.go
package main

func kthDistinct(arr []string, k int) string {
	strings := map[string]bool{}
	for _, str := range arr {
		if _, ok := strings[str]; ok {
			strings[str] = false
		} else {
			strings[str] = true
		}
	}
	count := 0
	for _, str := range arr {
		if strings[str] {
			count++
			if count == k {
				return str
			}
		}
	}
	return ""
}

File: LeetCode_ProblemSet/Problem2053.py
class Solution(object):
    def kthDistinct(self, arr, k):
        """
        :type arr: List[str]
        :type k: int
        :rtype: str
        """
        map = {}
        for str in arr:
            if str in map:
                map[str] = False
            else:
                map[str] = True
                
        cnt = 0
        for str in arr:
            if map[str]:
                cnt += 1
                if cnt == k:
                    return str
                
        return ""